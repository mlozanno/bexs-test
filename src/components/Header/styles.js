import styled from 'styled-components';
import { spacing, colors } from '~/styles/theme';

export const StyledHeader = styled.header`
	display: flex;
	align-items: center;
	justify-content: space-between;
	height: ${spacing(7)};
	padding: ${spacing(0, 2)};
	background-color: ${colors.light};
	border-bottom: 1px solid ${colors.darkGrey};

	img {
		display: block;
		max-height: 3rem;
	}
`;
