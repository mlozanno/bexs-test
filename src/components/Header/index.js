import React from 'react';

import { Link } from 'react-router-dom';

import Nav from '~/components/Nav';
import logo from '~/assets/logo.png';
import logo2x from '~/assets/logo@2x.png';

import { StyledHeader, Content } from './styles';

const Header = () => (
	<StyledHeader data-testid="header">
		<Content>
			<Link to="/">
				<img src={logo} srcSet={`${logo2x} 2x`} alt="Bexs logo" />
			</Link>
			<Nav />
		</Content>
	</StyledHeader>
);

export default Header;
