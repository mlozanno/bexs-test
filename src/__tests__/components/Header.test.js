import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import '@testing-library/jest-dom';

import Header from '~/components/Header';

describe('<Header />', () => {
	it('should be a function', () => {
		expect(typeof Header).toBe('function');
	});

	it('should render component', () => {
		const { getByTestId } = render(
			<MemoryRouter>
				<Header />
			</MemoryRouter>
		);

		const header = getByTestId('header');

		expect(header).toMatchSnapshot();
	});
});
