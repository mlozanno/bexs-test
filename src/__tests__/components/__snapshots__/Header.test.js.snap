// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render component 1`] = `
<header
  class="sc-AxhCb iuxROu"
  data-testid="header"
>
  <div
    class="sc-AxhUy gcCRfg"
  >
    <a
      href="/"
    >
      <img
        alt="Bexs logo"
        src="logo.png"
        srcset="logo@2x.png 2x"
      />
    </a>
    <button
      class="sc-AxirZ xiPux"
      data-testid="toggle-button"
      type="button"
    >
      <svg
        fill="currentColor"
        height="40"
        size="40"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
    <nav
      class="sc-AxjAm bDrSHG"
    >
      <ul
        class="sc-AxiKw kuWfqk"
        data-testid="nav"
      >
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/checkout"
          >
            Checkout
          </a>
        </li>
        <li>
          <a
            href="/carteira"
          >
            Carteira
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
